lib/models/task-dependency.js

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = TaskDependencyFactory;
+
+TaskDependencyFactory.$provide = 'Models.TaskDependency';
+TaskDependencyFactory.$inject = [
+    'Model',
+    'Constants'
+];
+
+function TaskDependencyFactory (Model, Constants) {
+    return Model.extend({
+        connection: 'mongo',
+        identity: 'taskdependencies',
+        attributes: {
+            domain: {
+                type: 'string',
+                defaultsTo: Constants.DefaultTaskDomain
+            },
+            graphId: {
+                type: 'string',
+                required: true
+            },
+            state: {
+                type: 'string',
+                required: true
+            },
+            evaluated: {
+                type: 'boolean',
+                defaultsTo: false
+            },
+            reachable: {
+                type: 'boolean',
+                defaultsTo: true
+            },
+            taskRunnerLease: {
+                type: 'string',
+                defaultsTo: null
+            },
+            taskRunnerHeartbeat: {
+                type: 'date',
+                defaultsTo: null
+            },
+            dependencies: {
+                type: 'json',
+                required: true
+            },
+            terminalOnStates: {
+                type: 'array',
+                defaultsTo: []
+            },
+            context: {
+                type: 'json',
+                defaultsTo: null
+            },
+            toJSON: function() {
+                // Remove waterline keys that we don't want in our dependency object
+                var obj = this.toObject();
+                delete obj.createdAt;
+                delete obj.updatedAt;
+                delete obj.id;
+                return obj;
+            }
```

Comment:
We also do this in models.graphobjects. These are Waterline internal fields that we don't expose to code that accesses this data structure. `toJSON` is called before any documents are returned from waterline query methods.